packages/portfolio-contract/tools/plan-solve.ts

```diff
@@ -388,6 +392,130 @@ export const solveRebalance = async (
   return { flows, detail: { solution } };
 };
 
+/**
+ * Compute partial order from solved flows and initial supplies.
+ *
+ * For each node, we track available supply (initial + completed inflows).
+ * An outflow step depends on the minimum set of inflow steps needed to
+ * provide sufficient supply.
+ *
+ * This approach correctly handles cases like:
+ * - Multiple operations to/from the same node with separate dependencies
+ * - Initial supplies that satisfy some outflows immediately
+ * - Fan-out patterns where multiple outflows depend on the same inflow
```

Comment:
I think this comment is backed by tests, but I don't know that we have any maintenance triggers to keep them in sync.

for the zillionth time, I [wish for strong doctest norms in the JS community](https://www.diigo.com/user/dckc-madmode?query=%23javascript+doctest).